components/site/src/lib.rs

```diff
@@ -864,9 +864,44 @@ impl Site {
         context.insert("taxonomies", &taxonomies);
         context.insert("config", &self.config);
 
-        let sitemap = &render_template("sitemap.xml", &self.tera, context, &self.config.theme)?;
+        // Count total number of urls to include in sitemap
+        let total_number = pages.len() + sections.len() + taxonomies.len();
+        if total_number > self.config.sitemap_limit {
+            // Split the sitemap and reference all sitemaps in a main sitemap
+
+            // Group all sitemap entries in one vector
+            let mut all_sitemap_entries = Vec::new();
+            all_sitemap_entries.append(&mut pages);
+            all_sitemap_entries.append(&mut sections);
+            for terms in taxonomies {
+                let mut terms = terms;
+                all_sitemap_entries.append(&mut terms);
+            }
+            
+            // Slice the vector in chunks and create sitemap file for each
+            let mut xml_files = Vec::new();
+            for (i, chunk) in all_sitemap_entries.chunks(self.config.sitemap_limit).enumerate() {
+                let mut chunk_context = context.clone();
+                chunk_context.insert("chunk", &chunk);
+                let sitemap = &render_template("multi_sitemap.xml", &self.tera, chunk_context, &self.config.theme)?;
+                let file_name:String = format!("sitemap{}.xml", i+1);
+                create_file(&self.output_path.join(&file_name), sitemap)?;
+                let mut xml_link:String = self.config.make_permalink(&file_name);
+                xml_link.pop(); // Remove trailing slash
+                xml_files.push(xml_link);
+            }
 
-        create_file(&self.output_path.join("sitemap.xml"), sitemap)?;
+            // Create main sitemap
+            let mut main_context = context.clone();
+            main_context.insert("xml_files", &xml_files);
```

Comment:
Not sure about `xml_files`, how about `sitemaps`?

Reply:
there is a "sitemap" variable just above so it might be confusing: 
`let sitemap = &render_template("split_sitemap.xml", &self.tera, context, &self.config.theme)?;`
Maybe "sub_sitemap".